app/app.py

```diff
@@ -1,51 +1,63 @@
 """
 Dockless origin/destination trip data API
+
+# try me
+http://localhost:8000/v1/trips?xy=-97.75094341278084,30.276185988411257&flow=destination
+
 """
 import argparse
 import json
 import os
-import pdb
 
-from flask import Flask, jsonify
-from flask_restful import Resource, Api, reqparse
-from flask_cors import CORS
+from sanic import Sanic
+from sanic import response
+from sanic import exceptions
+from sanic_cors import CORS, cross_origin
 from shapely.geometry import Point, shape, asPolygon, mapping
 from shapely.ops import cascaded_union
 
 dirname = os.path.dirname(__file__)
 source = os.path.join(dirname, "data/grid.json")
 
 with open(source, "r") as fin:
-
     data = json.loads(fin.read())
+    app = Sanic(__name__)
+    CORS(app)
 
-    parser = reqparse.RequestParser()
 
-    parser.add_argument(
-        "xy",
-        required=True,
-        type=str,
-        help="Coordinate string missing or unable to convert to decimal value. Must be formatted as comma-separated string in format x,y,x,y...",
-    )
-    
-    parser.add_argument(
-        "mode",
-        type=str,
-        choices=["origin", "destination"],
-        help="Mode string must be one of (origin, destination)",
-    )
 
-    app = Flask(__name__)
-    api = Api(app)
-    cors = CORS(app, resources={r"/api/*": {"origins": "*"}})
+def parse_flow(args):
+    if not args.get("flow") or args.get("flow") == "origin":
+        return "origin"
+    elif args.get("flow") == "destination":
+        return "destination"
+    else:
+        raise exceptions.ServerError("Unsupported flow specified", status_code=500)
+
 
+def parse_coordinates(args):
+    if not args.get("xy"):
+        raise exceptions.ServerError("XY parameter is requried.", status_code=500)
 
-def parse_coordinates(xy_string):
-    elements = xy_string.split(",")
-    elements = [float(elem) for elem in elements]
+    elements = args.get("xy").split(",")
```

Comment:
don't think this matters too much and your the geo expert but why not accept/require x & y as two separate parameters instead of as a comma-separated string? Not sure what most geoapps do.